File: Lab/Lab1_CPP/parrot.cpp
#include <iostream>   // oba rozwiazania poprawne
#include <vector>
#include <string>
#include <cstdlib>
#include <ctime>

using namespace std;

class Parrot {
private:
    vector<string> phrase;
public:
    Parrot(string Phrase) {
        phrase.push_back(Phrase);
    }

    void say(int repetition = 1) {
        if (!phrase.empty()) {
            for (int i = 0; i < repetition; i++) {
                int random = rand() % phrase.size();
                cout << phrase[random] << endl;
            }
        }
    }

    void addPhrase(string newPhrase) {

        phrase.push_back(newPhrase);

    }


};


int main() {

    srand(time(0));
    Parrot Parrot("Hello World"); 
    Parrot.say(1);


    Parrot.addPhrase("Hello World");
    Parrot.addPhrase("Hello");
    
    Parrot.say(1);
    Parrot.say(1);
    return 0;
}

File: Lab/Lab1_Java/src/Parrot.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Parrot {
	
	private List<String> Phrase;
	private Random random;
	
	public Parrot() {
		this.Phrase = new ArrayList<>();
		this.random = new Random();
	}
	
	public void addPhrase(String phrase) {
		Phrase.add(phrase);
	}
	
	private void say(int repetition) {
		String choosePhrase = Phrase.get(random.nextInt(Phrase.size()));
		for (int i = 0; i < repetition; i++) {
			System.out.println(choosePhrase);
		}
	}
	
	public static void main(String[] args) {

		Parrot parrot = new Parrot();
		
		parrot.addPhrase("Hello World");
		parrot.addPhrase("Hello");
		
		parrot.say(2);
	}

}
